lib/event-target.js

```diff
@@ -111,13 +111,15 @@ const EventTarget = {
    *
    * @param {String} method A string representing the event type to listen for
    * @param {Function} listener The listener to add
+   * @param {Object} [options] An options object specifies characteristics about the event listener
    * @public
    */
-  addEventListener(method, listener) {
+  addEventListener(method, listener, options = {}) {
     if (typeof listener !== 'function') return;
 
     function onMessage(data) {
       listener.call(this, new MessageEvent(data, this));
+      if (options.once) this.removeListener(method, listener);
```

Comment:
I think it's better to add it back also for other events to ensure that the listener is removed as documented. This should also be supported for custom events (line 180).

Not sure if it's better to remove it before the listener is called or after as it is currently done. Does the browser removes it before or after?

Reply:
no idea of what the browser do but it seems indeed better to remove it first to avoid any same tick problem